transport/http/outbound.go

```diff
@@ -282,7 +282,7 @@ func (o *Outbound) callWithPeer(
 		// maintenance loop resumes probing for availability.
 		p.OnDisconnected()
 
-		return nil, err
+		return nil, yarpcerrors.UnknownErrorf("unknown error from http client: %s", err.Error())
```

Comment:
Are we 100% sure that this is an Unavailable error though? If we're not sure, when we add connection retries to outbounds it's possible that we might accidentally retry requests that are not idempotent.